src/functions/onDisconnect.ts

```diff
@@ -15,13 +45,19 @@ export const handler: Handler = async (event: WebSocketAPIGatewayEvent): Promise
   const { connectionId } = event.requestContext;
 
   console.log('Deleting connectionId from the db table...');
+  const ws = new WebSocketClient(event.requestContext);
   try {
     // Delete user from ConnectionsTable
-    await deleteConnection(connectionId);
+    const deletedUser = await deleteConnection(connectionId);
+
+    // Additional cleanup when user disconnect
+    if (deletedUser) {
+      await cleanupUserWhenDisconnect(ws, deletedUser);
+    }
 
     return response(200, 'Connection deleted successfully');
   } catch (err) {
```

Comment:
We are forgetting to send USER_UPDATE to other users connected to WS to remove them from user list.
{action: "USER_UPDATE", payload: {user, state: "DISCONNECTED"}

Reply:
not sure if I understand this. so, no matter the game is started or not, whenever the host leaves the game, we want to send this
```
{action: "USER_UPDATE", payload: {user, state: "DISCONNECTED"}
```
to kick all the users out of the game?

Reply:
This is unrelated to the game. We want to remove the user from user list in clients when they disconnect